src/main/java/com/minecolonies/coremod/entity/ai/citizen/cook/EntityAIWorkCook.java

```diff
@@ -117,35 +124,56 @@ public EntityAIWorkCook(@NotNull final JobCook job)
         super.registerTargets(
                 new AITarget(IDLE, START_WORKING),
                 new AITarget(START_WORKING, this::startWorking),
-                new AITarget(COOK_GATHERING, this::gatherFoodFromBuilding),
+                new AITarget(COOK_GATHER_FOOD_FROM_BUILDING, this::gatherFoodFromBuilding),
                 new AITarget(COOK_COOK_FOOD, this::cookFood),
-                new AITarget(COOK_SERVE, this::serveFood),
-                new AITarget(COOK_RETRIEVE_FOOD, this::retrieve),
-                new AITarget(COOK_GET_FIREWOOD, this::getBurnableMaterial)
+                new AITarget(COOK_SERVE_FOOD_TO_CITIZEN, this::serveFoodToCitizen),
+                new AITarget(COOK_GATHER_COOKED_FOOD_FROM_FURNACE, this::gatherCookedFoodFromFurnace),
+                new AITarget(COOK_GATHER_FUEL, this::gatherFuel)
         );
         worker.setSkillModifier(CHARISMA_MULTIPLIER * worker.getCitizenData().getCharisma()
                 + INTELLIGENCE_MULTIPLIER * worker.getCitizenData().getIntelligence());
         worker.setCanPickUpLoot(true);
     }
 
-    private AIState getBurnableMaterial()
+    /**
+     * Find Fuel.
+     *
+     * If not walking and we need to walk to building, repeat this state.
+     * If building has fuel or is requesting fuel, wait a while, then COOK_COOK_FOOD
+     * If no fuel in building, transition to START_WORKING.
+     * If fuel in building but not nearby, walk toward it and repeat this state.
+     * If near fuel in building, pick up fuel.
+     * If fuel was no longer available, transition to START_WORKING.
+     * If we have fuel, then COOK_COOK_FOOD
+     *
+     * @return next AIState
+     */
+    private AIState gatherFuel()
     {
-        if(walkTo == null && walkToBuilding())
+        worker.setLatestStatus(new TextComponentTranslation(TranslationConstants.COM_MINECOLONIES_COREMOD_STATUS_RETRIEVING),
+            new TextComponentTranslation(":"),
+            new TextComponentTranslation(TranslationConstants.COM_MINECOLONIES_REQUESTS_BURNABLE));
+
+        // TODO: check if we already have the fuel in our inventory. If so, transition to COOK_FOOD immediately.
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)